### PR TITLE
Adding recursive option for cloning

### PR DESCRIPTION
**Pull request by rafecolton**
Addresses https://github.com/modcloth/kamino/issues/3

### CODE REVIEW

**Comment by rafecolton** [comment]
@jszwedko mind taking a look?

Turns out this was failing because the submodule that was being used for testing the `--recursive` option needed to be changed from ssh to https.

File: clone_creator.go [hunk 1]
```diff
@@ -79,21 +79,23 @@ func (creator *clone) cloneRepo(dest string) error {
...
+	if creator.Depth != "" {
+		cloneArgs = append(cloneArgs, "--depth", creator.Depth)
+	}
+
+	cloneArgs = append(cloneArgs, repoURL.String(), dest)
+
+	cloneCmd = &exec.Cmd{
+		Path:   git,
+		Args:   cloneArgs,
+		Stdout: buff,
```
Comment (jszwedko) [inline clone_creator.go:97]: Does using `buff` for both here risk interleaved output (assuming the kernel is buffering stdout/stderr)?

File: clone_creator.go [hunk 2]
```diff
@@ -140,10 +142,10 @@ func (creator *clone) cloneRepo(dest string) error {
 func (creator *clone) updateToRef(dest string) error {
 	/*
 		workflow as follows:
-			git clean -d --force --quiet
-			git fetch --prune --quiet
-			git checkout --force --quiet <ref>
-			git symbolic-ref --quiet HEAD && git pull --rebase --quiet
+			git clean -d --force
+			git fetch --prune
```
Comment (jszwedko) [inline clone_creator.go:146]: :+1: to removing `--quiet`

**Comment by jszwedko** [comment]
:shipit: aside from stdout/stderr question.

File: clone_creator.go (hunk 1 above)
Comment (rafecolton) [inline clone_creator.go:97]: Yeah, I don't know why I thought there needed to only be one stream.  Updated.

**Pull request merged**